lib/ngx/process.lua

```diff
@@ -0,0 +1,35 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+local getfenv = getfenv
+
+
+local C = ffi.C
+local _M = { version = base.version }
+
+
+ffi.cdef[[
+void ngx_http_lua_ffi_worker_privileged(int enable);
+int ngx_http_lua_ffi_worker_type(void);
+]]
+
+
+function _M.type()
+    return C.ngx_http_lua_ffi_worker_type()
+end
+
+
+function _M.privilege(enable)
```

Comment:
@membphis We need verbs in the method name. That's why I suggested `enable_privileged_agent()`. Otherwise it may seem like a method used to fetch info about the privileged agent process.